shotover-proxy/src/transforms/cassandra/peers_rewrite.rs

```diff
@@ -0,0 +1,321 @@
+        compression::Compression,
+        consistency::Consistency,
+        frame::{
+            frame_result::{
+                ColSpec,
+                ColType::{Inet, Int},
+                ColTypeOption, RowsMetadata, RowsMetadataFlags, TableSpec,
+            },
+            Direction, Frame as CassandraProtocolFrame, Opcode, Serialize,
+            {frame_query::BodyReqQuery, Flags, Version},
+        },
+        query::QueryParams,
+    };
+
+    fn create_query_message(query: String) -> Message {
+        let body = BodyReqQuery {
+            query,
+            query_params: QueryParams {
+                consistency: Consistency::One,
+                with_names: false,
+                values: None,
+                page_size: Some(5000),
+                paging_state: None,
+                serial_consistency: None,
+                timestamp: Some(1643855761086585),
+            },
+        };
+
+        let frame = CassandraProtocolFrame {
+            version: Version::V4,
+            direction: Direction::Request,
+            flags: Flags::default(),
+            stream_id: 0,
+            opcode: Opcode::Query,
+            tracing_id: None,
+            body: body.serialize_to_vec(),
+            warnings: vec![],
+        };
+
+        let bytes = frame.encode_with(Compression::None).unwrap();
+
+        Message {
+            details: MessageDetails::Unknown,
+            return_to_sender: false,
+            modified: false,
+            original: Frame::Cassandra(CassandraFrame::from_bytes(bytes.into()).unwrap()),
+        }
+    }
+
+    fn create_response_message(rows: Vec<Vec<MessageValue>>) -> Message {
+        let original = Frame::Cassandra(CassandraFrame {
+            version: Version::V4,
+            stream_id: 0,
+            tracing_id: None,
+            warnings: vec![],
+            operation: CassandraOperation::Result(Rows {
+                value: MessageValue::Rows(rows),
+                metadata: RowsMetadata {
+                    flags: RowsMetadataFlags::GLOBAL_TABLE_SPACE,
+                    columns_count: 1,
+                    paging_state: None,
+                    global_table_spec: Some(TableSpec {
+                        ks_name: "system".into(),
+                        table_name: "peers_v2".into(),
+                    }),
+                    col_specs: vec![ColSpec {
+                        table_spec: None,
+                        name: "native_port".into(),
+                        col_type: ColTypeOption {
+                            id: Int,
+                            value: None,
+                        },
+                    }],
+                },
+            }),
+        });
+
+        Message {
+            details: MessageDetails::Unknown,
+            return_to_sender: false,
+            modified: false,
+            original,
+        }
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_is_system_peers() {
```

Comment:
Why are these async? I dont see any awaits

Reply:
There used to be async there but I forgot to remove it.